eclair-core/src/main/scala/fr/acinq/eclair/wire/ChannelCodecs.scala

```diff
@@ -156,28 +156,30 @@ object ChannelCodecs extends Logging {
       ("sentAfterLocalCommitIndex" | uint64overflow) ::
       ("reSignAsap" | bool8)).as[WaitingForRevocation]
 
-  val localCodec: Codec[Origin.Local] = (
-    ("id" | uuid) ::
-      ("sender" | provide(Option.empty[ActorRef]))
-    ).as[Origin.Local]
+  val localColdCodec: Codec[Origin.LocalCold] = (
+    ("id" | uuid)).as[Upstream.Local].as[Origin.LocalCold]
 
-  val relayedCodec: Codec[Origin.Relayed] = (
+  val localCodec: Codec[Origin.Local] = localColdCodec.xmap[Origin.Local](o => o: Origin.Local, o => Origin.LocalCold(o.upstream))
+
+  val relayedColdCodec: Codec[Origin.ChannelRelayedCold] = (
     ("originChannelId" | bytes32) ::
       ("originHtlcId" | int64) ::
       ("amountIn" | millisatoshi) ::
-      ("amountOut" | millisatoshi)).as[Origin.Relayed]
+      ("amountOut" | millisatoshi)).as[Upstream.ChannelRelayedCold].as[Origin.ChannelRelayedCold]
+
+  val relayedCodec: Codec[Origin.ChannelRelayed] = relayedColdCodec.xmap[Origin.ChannelRelayed](o => o: Origin.ChannelRelayed, o => Origin.ChannelRelayedCold(Upstream.ChannelRelayedCold(o.upstream.originChannelId, o.upstream.originHtlcId, o.upstream.amountIn, o.upstream.amountOut)))
+
+  val trampolineRelayedColdCodec: Codec[Origin.TrampolineRelayedCold] = (
+    listOfN(uint16, bytes32 ~ int64)).as[Upstream.TrampolineRelayedCold].as[Origin.TrampolineRelayedCold]
```

Comment:
nit: unnecessary parenthesis:
```suggestion
  val trampolineRelayedColdCodec: Codec[Origin.TrampolineRelayedCold] = 
    listOfN(uint16, bytes32 ~ int64).as[Upstream.TrampolineRelayedCold].as[Origin.TrampolineRelayedCold]
```

Reply:
Done with 89ffc887b.